packages/cli/test/integration/credentials.ee.test.ts

```diff
@@ -0,0 +1,270 @@
+import express from 'express';
+import { UserSettings } from 'n8n-core';
+import { Db } from '../../src';
+import { randomName, randomString } from './shared/random';
+import config from '../../config';
+import * as utils from './shared/utils';
+import type { CredentialPayload, SaveCredentialFunction } from './shared/types';
+import type { Role } from '../../src/databases/entities/Role';
+import type { User } from '../../src/databases/entities/User';
+import * as testDb from './shared/testDb';
+import { RESPONSE_ERROR_MESSAGES } from '../../src/constants';
+import { CredentialsEntity } from '../../src/databases/entities/CredentialsEntity';
+
+jest.mock('../../src/telemetry');
+
+let app: express.Application;
+let testDbName = '';
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+let credentialOwnerRole: Role;
+let saveCredential: SaveCredentialFunction;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({
+		endpointGroups: ['credentials'],
+		applyAuth: true,
+	});
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	utils.initConfigFile();
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+	credentialOwnerRole = await testDb.getCredentialOwnerRole();
+	saveCredential = affixRoleToSaveCredential(credentialOwnerRole);
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'SharedCredentials', 'Credentials'], testDbName);
+	config.set('deployment.paid', true);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+// ----------------------------------------
+// dynamic router switching
+// ----------------------------------------
+
+test('dynamic router switching', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
```

Comment:
tbh I think the specific naming with `owner` and `auth` is just needed if you have multiple agents in a test, otherwise it is just clunky naming without adding additional value

Reply:
We should change this elsewhere as well for consistency.

Reply:
Tech debt ticket: https://linear.app/n8n/issue/N8N-4072/make-agent-calls-consistent-in-cli-tests